### PR TITLE
add testDrawRepayDebtFuzzy test for ERC20 and ERC721 Pools

### PR DESCRIPTION
**Pull request by MikeHathaway**
- Adds `testDrawRepayDebtFuzzy` tests to `ERC20Borrow.t.sol` and `ERC721Borrow.t.sol` for the NFT collection.
- Add FuzzyHelperContracts for both pool types that skip chain forking to reduce test run time.
- Adds utility methods required to execute fuzz tests.
- Remove unused solmate library

### CODE REVIEW

File: tests/forge/utils/DSTestPlus.sol [hunk 1]
```diff
@@ -1106,4 +1106,50 @@ abstract contract DSTestPlus is Test, IPoolEvents {
...
+        }
+    }
+
+    // calculate required collateral to borrow a given amount at a given limitIndex
+    function _requiredCollateral(uint256 borrowAmount, uint256 indexPrice) internal view returns (uint256 requiredCollateral_) {
+        // calculate the required collateral based upon the borrow amount and index price
+        (uint256 interestRate, ) = _pool.interestRateInfo();
+        uint256 newInterestRate = Maths.wmul(interestRate, 1.1 * 10**18); // interest rate multipled by increase coefficient
+        uint256 expectedDebt = Maths.wmul(borrowAmount, _feeRate(newInterestRate) + Maths.WAD);
+        requiredCollateral_ = Maths.wdiv(expectedDebt, _poolUtils.indexToPrice(indexPrice)) + Maths.WAD;
```
Comment (prat-gpt) [inline tests/forge/utils/DSTestPlus.sol]: We have added `Maths.WAD` here to make sure that borrow doesn't fail due to precision loss in collateral amount?

Comment (MikeHathaway) [reply tests/forge/utils/DSTestPlus.sol]: That was unneeded, leftover from a previous implementation. I removed it

**Comment by MikeHathaway** [comment]
The additional fuzz tests occasionally take a significant amount of time to run. It may be worth marking them as `testLoad` to simplify the development experience.

File: src/base/Pool.sol [hunk 2]
```diff
@@ -336,25 +336,25 @@ abstract contract Pool is Clone, ReentrancyGuard, Multicall, IPool {
     /*****************************/
 
     function _accruePoolInterest() internal returns (PoolState memory poolState_) {
-	// retrieve t0Debt amount from poolBalances struct
+	    // retrieve t0Debt amount from poolBalances struct
         uint256 t0Debt = poolBalances.t0Debt;
 
-	// initialize fields of poolState_ struct with initial values
+	    // initialize fields of poolState_ struct with initial values
```
Comment (MikeHathaway) [inline src/base/Pool.sol:342]: Might be an editor configuration issue here - these were incorrectly offset for me locally.

**Review by grandizzy — APPROVED** [review]
:+1:

**Pull request merged**